pkg/channel/distributed/dispatcher/dispatcher/handler_test.go

```diff
@@ -203,26 +227,34 @@ func performHandlerConsumeClaimTest(t *testing.T, destinationUri, replyUri, dead
 	// Create The Handler To Test
 	handler := createTestHandler(t, destinationUri, replyUri, &deliverySpec)
 
+	consumeFinishedChan := make(chan bool)
 	// Background Start Consuming Claims
 	go func() {
 		err := handler.ConsumeClaim(mockConsumerGroupSession, mockConsumerGroupClaim)
 		assert.Nil(t, err)
+		consumeFinishedChan <- true
 	}()
 
 	// Perform The Test (Add ConsumerMessages To Claims)
 	consumerMessage := createConsumerMessage(t)
 	mockConsumerGroupClaim.MessageChan <- consumerMessage
 
-	// Wait For Message To Be Marked As Complete
-	markedMessage := <-mockConsumerGroupSession.MarkMessageChan
+	if !expectSkipMark {
+		// Wait For Message To Be Marked As Complete
+		markedMessage := <-mockConsumerGroupSession.MarkMessageChan
+
+		// Verify The Results (CloudEvent Was Dispatched & ConsumerMessage Was Marked)
+		assert.Equal(t, consumerMessage, markedMessage)
+		assert.NotNil(t, mockMessageDispatcher.Message())
+		verifyDispatchedMessage(t, mockMessageDispatcher.Message())
+	}
+	assert.Equal(t, !expectSkipMark, mockConsumerGroupSession.MarkMessageCalled())
 
 	// Close The Mock ConsumerGroupClaim Message Channel To Complete/Exit Handler's ConsumeClaim()
 	close(mockConsumerGroupClaim.MessageChan)
 
-	// Verify The Results (CloudEvent Was Dispatched & ConsumerMessage Was Marked)
-	assert.Equal(t, consumerMessage, markedMessage)
```

Comment:
can you put this `// Verify The Results` comment up above new section line 242?

Reply:
et voilà